
Erlang 14B01 (aka 14.2.1 in Gentoo) fails with 64bit arches and FORTIFY_SOURCE=2 or greater, consisting of a buffer overflow warning, which is not an overflow but an overreaction of GCC.

Upstream report: http://www.erlang.org/cgi-bin/ezmlm-cgi?2:mss:2135:bdnooaggpdnbokaboakj
Upstream fix: https://github.com/erlang/otp/commit/3b7fd1030711ad56fa45d18dc6e0f1d0beee2399

http://bugs.gentoo.org/show_bug.cgi?id=348559

--- erts/emulator/drivers/common/efile_drv.c
+++ erts/emulator/drivers/common/efile_drv.c
@@ -385,7 +385,6 @@
 	    ErlDrvBinary *binp;
 	    int           size;
 	    int           offset;
-	    char          name[1];
 	} read_file;
 	struct {
 	    struct t_readdir_buf *first_buf;
@@ -1117,7 +1116,7 @@
 	Sint64 size;
 	
 	if (! (d->result_ok = 
-	       efile_openfile(&d->errInfo, d->c.read_file.name, 
+	       efile_openfile(&d->errInfo, d->b, 
 			      EFILE_MODE_READ, &fd, &size))) {
 	    goto done;
 	}
@@ -3071,7 +3070,7 @@
 	d->command = command;
 	d->reply = !0;
 	/* Copy name */
-	FILENAME_COPY(d->c.read_file.name, filename);
+	FILENAME_COPY(d->b, filename);
 	d->c.read_file.binp = NULL;
 	d->invoke = invoke_read_file;
 	d->free = free_read_file;
